holoviews/plotting/bokeh/heatmap.py

```diff
@@ -71,60 +79,105 @@ def get_data(self, element, ranges, style):
         x, y, z = (dimension_sanitizer(d) for d in element.dimensions(label=True)[:3])
         if self.invert_axes: x, y = y, x
         cmapper = self._get_colormapper(element.vdims[0], element, ranges, style)
-        if 'line_alpha' not in style and 'line_width' not in style:
-            style['line_alpha'] = 0
-            style['selection_line_alpha'] = 0
-            style['nonselection_line_alpha'] = 0
-        elif 'line_color' not in style:
-            style['line_color'] = 'white'
 
         if not element._unique:
             self.param.warning('HeatMap element index is not unique,  ensure you '
                                'aggregate the data before displaying it, e.g. '
                                'using heatmap.aggregate(function=np.mean). '
                                'Duplicate index values have been dropped.')
 
-        if self.static_source:
+        is_gridded = element.interface.gridded
+        x_index = y_index = None
+        if is_gridded:
+            x_range, y_range = self.handles['x_range'], self.handles['y_range']
+            x_cat, y_cat = isinstance(x_range, FactorRange), isinstance(y_range, FactorRange)
+            if x_cat:
+                xs = self._get_dimension_factors(element, ranges, element.get_dimension(x))
+                x_index = find_contiguous_subarray(xs, x_range.factors)
```

Comment:
I think we can take advantage of the fact that the categories do not have any duplicates (or am I missing something?) Something like this (missing some of the checks)

``` python 
x = np.asarray([np.str_('A'), np.str_('B'), np.str_('C'), np.str_('D'), np.str_('E'), np.str_('F'), np.str_('G'), np.str_('H'), np.str_('I'), np.str_('J')])
y = np.asarray([ np.str_('E'), np.str_('F'), np.str_('G')])


def example(x, y):
    cmp = np.asarray(x) == y[0]
    
    if not cmp.any():
        return None
    
    idx = np.argmax(cmp)
    
    if not (x[idx:idx+len(y)] == y).all():
        return None
    
    return idx

example(x, y)

```

Reply:
True, that should work.

Reply:
Okay, swapped it out.